### PR TITLE
feat: P4-3891 Add a lodge form

### PR DESCRIPTION
**Pull request by Thource**
### What changed

A form has been added to add planned lodges to a move.

### Why did it change

This has been added so that BaSM users can plan lodges for a move.

### Issue tracking

- P4-3891

## Screenshots

| Before | After |
| ------ | ----- |
| <kbd></kbd> | <kbd></kbd> |

#### Automated testing

- [ ] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [ ] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/main/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [ ] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

**Review by jimbali — APPROVED** [review]
lgtm

**Review by danbenton-mojdt — DISMISSED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -18,19 +18,8 @@ module.exports = {
...
-    'comma-dangle': [
-      'error',
-      {
-        arrays: 'always-multiline',
-        objects: 'always-multiline',
-        imports: 'never',
-        exports: 'never',
-        functions: 'ignore',
-      },
-    ],
```
[the remark this answers is not in the document]
Comment (Thource) [reply .eslintrc.js:30]: It gets rid of the conflict between eslint and prettier :)

**Review by jimbali — DISMISSED** [review]
lgtm except for one small comment

File: app/move/app/add-lodge/controllers/save.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  async successHandler(req: any, res: any, next: any) {
+    const dateFormat = 'yyyy-MM-dd'
+
+    const lodgeLocation = req.sessionModel.attributes.to_location_lodge
+    const startDate = parseISO(req.sessionModel.attributes.lodge_start_date)
+    const lodgeLength = Number(req.form.values.lodge_length)
+    const endDate = new Date(startDate.getTime())
+    endDate.setDate(endDate.getDate() + lodgeLength)
```
[the remark this answers is not in the document]
Comment (Thource) [reply app/move/app/add-lodge/controllers/save.ts:19]: It converts the start timestamp to a date, clones it and then adds lodgeLength days by using setDate getDate, but I'll look into date-fns/add instead

**Review by jimbali — APPROVED** [review]
lgtm!

**Pull request merged**